src/circular_buffer.jl

```diff
@@ -28,17 +30,11 @@ function Base.empty!(cb::CircularBuffer)
     return cb
 end
 
-Base.@propagate_inbounds function _buffer_index_checked(cb::CircularBuffer, i::Int)
-    @boundscheck if i < 1 || i > cb.length
-        throw(BoundsError(cb, i))
-    end
-    _buffer_index(cb, i)
-end
```

Comment:
This can be collapsed into `_buffer_index`, which can be simplified without loss of performance. Callers can use `@inbounds` to enable or disable bounds checking.